### DISCUSSION TITLE
Pydantic "hide_input_in_errors" does not hide "input" field when request validation fails on model

### DISCUSSION THREAD

**Discussion by progsmile** [fastapi/fastapi#10025] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
Follow reproduce steps within repo:
https://github.com/progsmile/fast-api-pydantic-bug
```

### Description

Pydantic "hide_input_in_errors" does not hide "input" field when request validation fails on model. Expected behaviour: there is no `input` field in response.

### Operating System

Linux

### FastAPI Version

0.101.0

### Pydantic Version

2.1.1

### Python Version

3.10.12

**Comment by anton-koltsov-veeva** [comment]
I confirm this happens on:
```
fastapi==0.101.1
pydantic==2.3.0
pydantic-settings==2.0.3
pydantic_core==2.6.3
```

**Comment by samuelpedrajas** [comment]
This is relevant: https://github.com/pydantic/pydantic/issues/7461

FastAPI is probably calling either the `errors()` or `json()` methods internally, and since these methods have the parameter `include_input` defaulted to True, the `hide_input_in_errors` configuration is ignored.

**Reply by progsmile** [reply]
Thanks! Does new pydantic fixes issue?

**Comment by YuriiMotov** [comment]
This config flag is for printing only and not respected by `e.json()` and `e.errors()` in Pydantic.
https://github.com/pydantic/pydantic/issues/7461#issuecomment-1721619883

In comments there has been expressed the idea of using this config value as a default value for `include_input` in `.json()` and `.errors()` methods. It would solve the problem, but it hasn't been implemented on Pydantic's side.
